turbine/src/broadcast_stage/broadcast_utils.rs

```diff
@@ -76,19 +100,19 @@ fn max_coalesce_time(serialized_batch_byte_count: u64, max_batch_byte_count: u64
 }
 
 pub(super) fn recv_slot_entries(
```

Comment:
I think it would be simpler for here and the shredding code if we didn't mix the components together.
Essentially change `ReceiveResults` to either return an entry batch or a block marker.

Then the shredding code becomes really simple as broadcast stage already does the scan to update the next_*_index and cmr - you just have to change entries_to_shreds to either call `make_merkle_shreds_from_entries` or `make_merkle_shreds_from_block_marker` (or just use make from data slice for both), and we can avoid the  tracking in `make_merkle_shreds_from_components`